runtime/cpp/cmake/SourcesAndHeaders.cmake

```diff
@@ -24,4 +24,5 @@ set(skel_includes
 
 set(test_sources
     src/bpf_api_test.cpp
+    src/memory_check_test_driver.cpp
```

Comment:
Keep the same format

Reply:
`memory_check_test_driver.cpp` is not a single test, it's a wasm program runner and will run two wasm programs as test. I made this to avoid writting two duplicated source files.